migrations/env.py

```diff
@@ -0,0 +1,57 @@
+import configparser
+import pathlib
+
+from sqlalchemy import engine_from_config, create_engine
+from sqlalchemy import pool
+
+from alembic import context
+from phabricatoremails import models
+
+target_metadata = models.Base.metadata
+config = configparser.ConfigParser()
+config.read(str(pathlib.Path(__file__).parent.parent / "settings.ini"))
```

Comment:
Hey, good eye! Yeah, somehow I need to inform `alembic` about the `settings.ini` location programatically so it can get the DB connection string. Thanks for spotting this.

> Usage of `parent.parent` here is not ideal and is repeated throughout the project.

Would you recommend `pathlib.Path(__file__) / "../../path"` instead, or is there a cleaner solution?

Reply:
>Would you recommend pathlib.Path(__file__) / "../../path" instead, or is there a cleaner solution?

At least I would place this somewhere else where you can import it, as this is repeated many times throughout the code base.